components/elm/src/main/ncdio_pio.F90.in

```diff
@@ -1888,10 +1897,15 @@ end subroutine ncd_io_{DIMS}d_{TYPE}_glob
           if (single_column) then
              start(:) = 1 
              count(:) = 1
-             call scam_field_offsets(ncid, elmlevel, vardesc, start, count)
+             call scam_field_offsets(ncid, elmlevel, vardesc, start, count,time_dim_exists = time_dim_exists)
              if (present(nt)) then
-               n=5
-               if (start(5) .eq. 0) n=4
+               if(time_dim_exists) then !if time dimesion is considered by scam, shift indices of start
+                  n=4
+                  if (start(4) .eq. 0) n=3
```

Comment:
If you fill up start(:) based on whether the time dim exists inside scam_field_offsets() you might not need to track that information (time_dim_exists) here, right? i.e., can you remove the time_dim_exists variable here and change the code to,
```
if (present(nt) then
  n = 5
  ! This could also be a while loop looking for the last zero val in start(1:5)
  if(start(n) .eq. 0) n = 4
  if(start(n) .eq. 0) n = 3
  start(n) = nt
  count(n) = 1  
end if
```

Reply:
Thanks @jayeshkrishna ! This is a great suggestion which will make the logic clearer. I will implement this logic and clear the time_dim_exists variable from the argument list.